### PR TITLE
RUN-2915 : Functional Tests covering Scheduled Jobs Workflows

### PR DESCRIPTION
**Pull request by mrdubr**
Functional tests that verify that Scheduled Jobs execute correctly after various types of changes

### CODE REVIEW

File: functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobExecutionSpec.groovy [hunk 1]
```diff
@@ -190,314 +190,6 @@ class JobExecutionSpec extends BaseContainer {
 
     }
 
-    def "test-job-flip-executionEnabled"(){
```
Comment (mrdubr) [inline functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobExecutionSpec.groovy:193]: Moved to `JobScheduledSpec` where scheduled job specs live

File: functional-test/src/test/resources/docker/compose/oss/docker-compose.yml [hunk 2]
```diff
@@ -34,6 +34,17 @@ services:
       - "./mock-scripts:/home/rundeck/mock-scripts:rw"
     ports:
       - "4440:4440"
+      - "5005:5005"
```
Comment (mrdubr) [inline functional-test/src/test/resources/docker/compose/oss/docker-compose.yml:37]: enables remote debugging and jmx access

File: functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobExecutionStatusSpec.groovy [hunk 3]
```diff
@@ -1,47 +1,140 @@
...
+            response.successful
+            response.code() == 200
+            def json = jsonValue(response.body())
+            json.executions.size() == 1
+            json.executions[0].id == execId
+        }
     }
 
-    def "job/id/executions?status=test+status+code with 1 results"() {
+    def "job/id/run should get into timedout status"() {
```
Comment (mrdubr) [inline functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobExecutionStatusSpec.groovy:48]: SIde-effects of refactoring.
Moved here from `JobScheduledSpec.groovy`

File: functional-test/src/test/groovy/org/rundeck/tests/functional/api/scm/ScmPluginImportScheduledJobSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import org.rundeck.util.api.scm.httpbody.GitImportSetupRequest
+import org.rundeck.util.api.scm.httpbody.ScmActionPerformRequest
+import org.rundeck.util.common.WaitingTime
+import org.rundeck.util.common.execution.ExecutionUtils
+import org.rundeck.util.common.jobs.JobUtils
+import org.rundeck.util.common.scm.ScmIntegration
+import org.rundeck.util.container.BaseContainer
+
+@APITest
+@ExcludePro
```
Comment (ahormazabal) [inline functional-test/src/test/groovy/org/rundeck/tests/functional/api/scm/ScmPluginImportScheduledJobSpec.groovy:16]: Why must this be excluded from pro?

**Review by ahormazabal — CHANGES_REQUESTED** [review]
LGTM but i need to get over two nits:
- Clarify why the test needs to be exluded from pro.
- Rebase the branch so it can pass the ci in rundeckpro.

**Review by ahormazabal — APPROVED** [review]

**Pull request merged**